tests/yam/agama/agama.pm

```diff
@@ -12,30 +12,21 @@ use testapi;
 
 use testapi qw(
   assert_script_run
-  assert_screen
   get_required_var
-  enter_cmd
-  select_console
 );
 
 sub run {
     my $self = shift;
     my $test = get_required_var('AGAMA_TEST');
+    my $reboot_page = $testapi::distri->get_reboot_page();
 
     script_run("dmesg --console-off");
     assert_script_run("/usr/share/agama/system-tests/" . $test . ".cjs", timeout => 1200);
     script_run("dmesg --console-on");
 
-    select_console 'displaymanager';
-    save_screenshot();
-
-    assert_screen('agama-install-finished', 10);
-    assert_and_click('reboot');
-}
-
-sub post_run_hook {
-    my ($self) = shift;
-    $self->SUPER::post_run_hook;
+    select_console('installation');
+    $reboot_page->expect_is_shown(timeout => 30);
```

Comment:
30 seconds is the default in openQA, no need to pass it.
I'm curious if `0` would work, as the screen is already there, could you try?